### PR TITLE
resource-previews

### PR DESCRIPTION
**Pull request by tarawren**
https://trello.com/c/qZOyrswB/13-section-resource-preview-large
https://trello.com/c/5sMuf9AC/22-section-resource-preview-small

1. Added `location` filters to the new `location` based resource sections (`small` and `large`)
2. Change `date` sort fields to descending for all `resource` views
3. Added `topic` taxonomy term and field to `resource` content type, as well as added the `exposed` `topic` filter to the main `resources` `view`
4. Added `views` and `paragraphs` for location-based `resource` sections
5. Updated styling

### CODE REVIEW

File: web/themes/fire/templates/views/resources-location-large/resources-location-large-filters.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    });
+  }
+
+  _labelClick() {
+    this.elements.$filterSelectListLabel.on('click', function () {
+      // show this dropdown
+      $(this).next('ul').slideToggle(200);
+
+      // hide other open dropdowns
+      $('.filter-select-list__label').not(this).next('ul').hide();
```
Comment (1forh) [inline web/themes/fire/templates/views/resources-location-large/resources-location-large-filters.js]: Could this be changed to a data attribute?

File: web/themes/fire/templates/views/resources-location-large/resources-location-large-filters.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      // add selected attr to matching select option and remove from siblings
+      $('.resources-location-small .form-select > option').each(function (id, el) {
+        if (el.value === selectedVal) {
+          $(this).siblings().removeAttr('selected');
+          $(this).attr('selected', 'selected');
+        }
+      });
+
+      // submit form after selection
+      $('.resources-location-large .form-submit').trigger('click');
```
Comment (1forh) [inline web/themes/fire/templates/views/resources-location-large/resources-location-large-filters.js]: Could this be changed to a data attribute?

File: web/themes/fire/templates/views/resources-location-small/resources-location-small-filters.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        });
+
+        $('.filter-select-list ul').hide();
+      });
+
+      // grab selected option text
+      var selectedLabel = $(this).find('option[selected]').text();
+
+      // pass selected option text into list label
+      $('.filter-select-list:last').children('.filter-select-list__label').text(selectedLabel);
```
Comment (1forh) [inline web/themes/fire/templates/views/resources-location-small/resources-location-small-filters.js]: Could this be changed to a data attribute?

File: web/themes/fire/templates/views/resources-location-small/resources-location-small-filters.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      var selectedVal = $(this).attr('value');
+
+      // pass selected option text into list label
+      $(this).parent().prev('.filter-select-list__label').text(selectedTxt);
+
+      // close dropdown after selection has been made
+      $(this).parent('ul').hide();
+
+      // add selected attr to matching select option and remove from siblings
+      $('.resources-location-small .form-select > option').each(function (id, el) {
```
Comment (1forh) [inline web/themes/fire/templates/views/resources-location-small/resources-location-small-filters.js]: Could this be changed to a data attribute?

File: web/themes/fire/templates/views/resources-location-small/resources-location-small-filters.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        if (el.value === selectedVal) {
+          $(this).siblings().removeAttr('selected');
+          $(this).attr('selected', 'selected');
+        }
+      });
+
+      // submit form after selection
+      $('.resources-location-small .form-submit').trigger('click');
```
Comment (1forh) [inline web/themes/fire/templates/views/resources-location-small/resources-location-small-filters.js]: Could this be changed to a data attribute?

**Comment by tarawren** [comment]
@1forh Updated!

**Pull request merged**